library/src/main/java/io/appwrite/services/Storage.kt

```diff
@@ -0,0 +1,265 @@
+    @Throws(AppwriteException::class)
+    suspend fun createFile(
+		file: File,
+		read: List<Any>? = null,
+		write: List<Any>? = null
+	): Response {
+        val path = "/storage/files"
+        val params = mapOf<String, Any?>(
+            "file" to file,
+            "read" to read,
+            "write" to write
+        )
+
+        val headers = mapOf(
+            "content-type" to "multipart/form-data"
+        )
+
+        return client.call("POST", path, headers, params)
+    }
+    
+    /**
+     * Get File
+     *
+     * Get a file by its unique ID. This endpoint response returns a JSON object
+     * with the file metadata.
+     *
+     * @param fileId
+     * @return The request response with a JSON body 
+     */
+    @JvmOverloads
+    @Throws(AppwriteException::class)
+    suspend fun getFile(
+		fileId: String
+	): Response {
+        val path = "/storage/files/{fileId}".replace("{fileId}", fileId)
+        val params = mapOf<String, Any?>(
+        )
+
+        val headers = mapOf(
+            "content-type" to "application/json"
+        )
+
+        return client.call("GET", path, headers, params)
+    }
+    
+    /**
+     * Update File
+     *
+     * Update a file by its unique ID. Only users with write permissions have
+     * access to update this resource.
+     *
+     * @param fileId
+     * @param read
+     * @param write
+     * @return The request response with a JSON body 
+     */
+    @JvmOverloads
+    @Throws(AppwriteException::class)
+    suspend fun updateFile(
+		fileId: String,
+		read: List<Any>?,
+		write: List<Any>?
+	): Response {
+        val path = "/storage/files/{fileId}".replace("{fileId}", fileId)
+        val params = mapOf<String, Any?>(
+            "read" to read,
+            "write" to write
+        )
+
+        val headers = mapOf(
+            "content-type" to "application/json"
+        )
+
+        return client.call("PUT", path, headers, params)
+    }
+    
+    /**
+     * Delete File
+     *
+     * Delete a file by its unique ID. Only users with write permissions have
+     * access to delete this resource.
+     *
+     * @param fileId
+     * @return The request response with a JSON body 
+     */
+    @JvmOverloads
+    @Throws(AppwriteException::class)
+    suspend fun deleteFile(
+		fileId: String
+	): Response {
+        val path = "/storage/files/{fileId}".replace("{fileId}", fileId)
+        val params = mapOf<String, Any?>(
+        )
+
+        val headers = mapOf(
+            "content-type" to "application/json"
+        )
+
+        return client.call("DELETE", path, headers, params)
+    }
+    
+    /**
+     * Get File for Download
+     *
+     * Get a file content by its unique ID. The endpoint response return with a
+     * 'Content-Disposition: attachment' header that tells the browser to start
+     * downloading the file to user downloads directory.
+     *
+     * @param fileId
+     * @return The request response with a JSON body 
```

Comment:
Response format is not JSON for this endpoint.

Reply:
Done :+1: